qkeras/quantizers.py

```diff
@@ -440,6 +441,555 @@ def non_trainable_variables(self):
+    def symmetric(self, symmetric):
+        self._symmetric = K.cast_to_floatx(symmetric)
+        if self._initialized:
+            self._calc_input_independent_attributes()
+
+    @property
+    def keep_negative(self):
+        return self._keep_negative
+
+    @keep_negative.setter
+    def keep_negative(self, keep_negative):
+        self._keep_negative = K.cast_to_floatx(keep_negative)
+        if self._initialized:
+            self._calc_input_independent_attributes()
+
+    @property
+    def qnoise_factor(self):
+        return self._qnoise_factor
+
+    @qnoise_factor.setter
+    def qnoise_factor(self, qnoise_factor):
+        self._qnoise_factor = K.cast_to_floatx(qnoise_factor)
+
+    @property
+    def use_stochastic_rounding(self):
+        return self._use_stochastic_rounding
+
+    @use_stochastic_rounding.setter
+    def use_stochastic_rounding(self, use_stochastic_rounding):
+        self._use_stochastic_rounding = tf.cast(use_stochastic_rounding, tf.bool)
+
+    @property
+    def scale_axis(self):
+        return self._scale_axis
+
+    @scale_axis.setter
+    def scale_axis(self, scale_axis):
+        self._scale_axis = scale_axis
+
+    @property
+    def alpha(self):
+        return self._alpha
+
+    @alpha.setter
+    def alpha(self, alpha):
+        """
+        Set alpha, _alpha_str, and auto_alpha attributes, and check if alpha is valid
+        Also, set scale if not auto_alpha.
+
+        Note: _alpha_str variable needed for uniform typing of alpha in tf.function
+        """
+
+        if alpha is None:
+            self._alpha = None
+            self._alpha_str = tf.cast('', tf.string)
+
+            # scale is always 1 for non-auto alpha
+            self.scale.assign(K.cast_to_floatx(1.0))
+            self.auto_alpha = tf.cast(False, tf.bool)
+        else:
+            # Check the quantizer has been given a valid alpha string
+            if not alpha in self.ALPHA_OPTIONS:
+                raise ValueError(
+                    f"Invalid alpha '{alpha}' for auto alpha computation. "
+                    f"Must be one of {self.ALPHA_OPTIONS}"
+                )
+            self._alpha = tf.cast(alpha, tf.string)
+            self._alpha_str = tf.cast(alpha, tf.string)
+            self.auto_alpha = tf.cast(True, tf.bool)
+
+    def _calc_input_independent_attributes(self):
+        """Calculate and set attributes that are independent of __call__ input"""
+        assert (
+            self._initialized
+        ), "Must initialize before calling _calc_input_independent_attributes"
+
+        if self.bits < self.integer + self.keep_negative:
+            err_msg = (
+                f"Bit count {self.bits} must exceed {self.integer + self.keep_negative}"
+            )
+            raise ValueError(err_msg)
+
+        # Get scale for integer representation, as given by parameters other than alpha
+        self.integer_repr_scale = K.pow(2.0, self.integer - self.bits + self.keep_negative)
```

Comment:
I'm not sure that `integer_repr_scale` is a very intuitive variable name. Have you considered other options?

Reply:
Updated to data_type_scale